### ISSUE TITLE
Investigate gossip communication within clusters

### ISSUE THREAD

**Issue by domodwyer** [influxdata/influxdb_iox#8201]
In production IOx runs as a clustered system of distinct processes - I believe we're missing a core communication primitive between these processes:

* https://github.com/influxdata/influxdb_iox/issues/6175
* https://github.com/influxdata/influxdb_iox/issues/5957
* https://github.com/influxdata/influxdb_iox/issues/5005
* https://github.com/influxdata/idpe/issues/17627#issuecomment-1587302228
* "Table delete" consistency in routers
* Probably a lot more that haven't been wrote, because it's just not been a possibility, for example:
  * Instead of the compactors polling the catalog for newly persisted files, inform them directly
  * Instead of the queriers discovering new parquet files during query execution, have them proactively informed and potentially cached / stats fetched
  * Instead of the ingesters discovering sort key updates after generating a parquet file (and having to throw it away) proactively inform them of sort key changes by other ingesters
  * etc etc etc - I think this is a "build it and they will come" kind of situation.

TL;DR - there's lots of opportunity for query / write performance gains, meaningful catalog load reductions, and better user-observable behaviours (like consistent service limits) if we can effectively communicate state changes between components on a best-effort basis without tight coupling and/or mesh networking (poor scalability).

I intend to try and solve some of the router issues linked above using a gossip primitive as a trial, and extend it to be used more widely if successfully.
